controllers/mirrorpeer_controller.go

```diff
@@ -161,9 +186,64 @@ func processMirrorPeerSecretChanges(ctx context.Context, rc client.Client, mirro
 	return anyErr
 }
 
+func (r *MirrorPeerReconciler) checkTokenExchangeStatus(ctx context.Context, mp multiclusterv1alpha1.MirrorPeer) (bool, error) {
+	logger := log.FromContext(ctx)
+	pr1 := mp.Spec.Items[0]
+	pr2 := mp.Spec.Items[1]
+
+	err := r.checkForSourceAndDestinationSecret(ctx, pr1, pr2.ClusterName)
+	if err != nil {
+		logger.Error(err, "Failed to find valid source and destination secret", "PeerRef", pr1)
+		return false, err
+	}
+
+	err = r.checkForSourceAndDestinationSecret(ctx, pr2, pr1.ClusterName)
+	if err != nil {
+		logger.Error(err, "Failed to find valid source and destination secret", "PeerRef", pr2)
+		return false, err
+	}
```

Comment:
Source or Destination? it should be clear.

Reply:
I have split into two funcs . Fixed